modules/dhcp/dhcp_api.rb

```diff
@@ -95,17 +131,48 @@ def load_subnet_data
     end
   end
 
-  # delete a record from a network
+  # deprecated, delete a record from a network
   delete "/:network/:record" do
     begin
       load_subnet
       load_subnet_data
 
+      logger.warn('DELETE dhcp/:network/:record endpoint has been deprecated and will be removed in future versions. '\
+                  'Please use DELETE dhcp/:network/mac/:mac_address or DELETE dhcp/:network/ip/:ip_address instead.')
+
       record = server.find_record(@subnet.network, params[:record])
-      log_halt 404, "DHCP record #{params[:network]}/#{params[:record]} not found" unless record
-      server.del_record @subnet, record
+      log_halt 404, "No DHCP record for #{params[:network]}/#{params[:record]} found" unless record
+      server.del_record(@subnet, record).to_json
+    rescue Proxy::DHCP::InvalidRecord
+      log_halt 404, "No DHCP records for #{params[:network]}/#{params[:record]} found"
+    rescue Exception => e
+      log_halt 400, e
+    end
+  end
+
+  # deletes all records for an ip address from a network
+  delete "/:network/ip/:ip_address" do
+    begin
+      load_subnet
+      load_subnet_data
+
+      server.del_records_by_ip(@subnet, params[:ip_address]).to_json
+    rescue Proxy::DHCP::InvalidRecord
+      log_halt 404, "No DHCP records for #{params[:network]}/#{params[:ip_address]} found"
```

Comment:
Agree re: load_subnet* methods, they are vestiges of the old design. I do think, however, that we shouldn't raise 404s on Proxy::DHCP::InvalidRecord because of reasons above -- the common exception handler is sufficient.

I think we are in agreement to handle deletion of missing records in a separate PR, returning 404s should be a part of that change.